core/string/ustring.cpp

```diff
@@ -1104,7 +1104,7 @@ const char32_t *String::get_data() const {
 	return size() ? &operator[](0) : &zero;
 }
 
-String String::_camelcase_to_underscore() const {
+String String::_separate_compound_words() const {
```

Comment:
Since this method is used in multiple places now, could you consider making it return a PackedStringArray and using the join method on String to concat with underscores/dashes/spaces? Not sure if it would be much of an improvement, but doing multiple replaces on the entire string feels bad.

Reply:
Sounds reasonable. I'm on it!

Reply:
Allright. I think the latest commit should address this suggestion. I also got rid of the multiple string replacements by just looping once over all characters and overwrite them if they match one of the criteria.

Now there is also the question of what to do with the c# side. It has it's own implementation of [SeparateCompoundWords](https://github.com/priahoud/godot/blob/0c0714fd98415efa0c71b7d14e82a63ef697f974/modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs#L387), which is only used in [Capitalize](https://github.com/priahoud/godot/blob/0c0714fd98415efa0c71b7d14e82a63ef697f974/modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs#L315). [ToPascalCase](https://github.com/priahoud/godot/blob/0c0714fd98415efa0c71b7d14e82a63ef697f974/modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs#L353), [ToCamelCase](https://github.com/priahoud/godot/blob/0c0714fd98415efa0c71b7d14e82a63ef697f974/modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs#L340), [ToSnakeCase](https://github.com/priahoud/godot/blob/0c0714fd98415efa0c71b7d14e82a63ef697f974/modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs#L366) and [ToKebabCase](https://github.com/priahoud/godot/blob/0c0714fd98415efa0c71b7d14e82a63ef697f974/modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs#L379) just get their result by marshalling it from the c++ side.

If we do the same with Capitalize, we could just get rid of the SeparateCompoundWords implementation on the  c# side. Overall simplifying things and at the same time making sure both sides behave the same.

Reply:
This new logic is very nice to follow! De-duplicating logic is of course always appreciated, but you'll have to talk to someone who knows more about the C# integration if there are any caveats when doing this, or intentionally different logic between the implementations.

Reply:
> I take it that your concerns are addressed now?

Yep, thanks!

> Ok, I will keep this conversation open and wait for someone from the C# side to comment.

You can also ask on the contributors chat, there is a `#dotnet` channel: https://chat.godotengine.org/